apps/api/migrations/preference-centralization/preference-centralization-migration.ts

```diff
@@ -0,0 +1,202 @@
+ * - subscriber workflow preferences
+ *    -> preferences with subscriber workflow type
+ * - workflow preferences
+ *   -> preferences with workflow-resource type
+ *   -> preferences with user-workflow type
+ */
+export async function preferenceCentralization(startWorkflowId?: string, startSubscriberId?: string) {
+  const app = await NestFactory.create(AppModule, {
+    logger: false,
+  });
+  console.log('start migration - preference centralization');
+
+  const upsertPreferences = app.get(UpsertPreferences);
+  const subscriberPreferenceRepository = app.get(SubscriberPreferenceRepository);
+  const workflowPreferenceRepository = app.get(NotificationTemplateRepository);
+
+  await migrateWorkflowPreferences(workflowPreferenceRepository, upsertPreferences, startWorkflowId);
+  console.log({ counter });
+  await migrateSubscriberPreferences(subscriberPreferenceRepository, upsertPreferences, startSubscriberId);
+
+  console.log('end migration - preference centralization');
+  console.log({ counter });
+  console.log(`Processed workflow preference with ID: ${lastProcessedWorkflowId}`);
+  console.log(`Processed subscriber preference with ID: ${lastProcessedSubscriberId}`);
+
+  app.close();
+}
+
+async function migrateWorkflowPreferences(
+  workflowPreferenceRepository: NotificationTemplateRepository,
+  upsertPreferences: UpsertPreferences,
+  startWorkflowId?: string
+) {
+  console.log('start workflow preference migration');
+  let query = {};
+  if (startWorkflowId) {
+    console.log(`Starting from workflow preference ID: ${startWorkflowId}`);
+    query = { _id: { $gt: startWorkflowId } };
+  }
+  const workflowPreferenceCursor = await workflowPreferenceRepository._model
+    .find(query)
+    .sort({ _id: 1 })
+    .batchSize(BATCH_SIZE)
+    .cursor();
+
+  for await (const workflowPreference of workflowPreferenceCursor) {
+    await sleep(10);
+    try {
+      const userWorkflowPreferenceToUpsert = UpsertUserWorkflowPreferencesCommand.create({
+        userId: workflowPreference._creatorId.toString(),
+        templateId: workflowPreference._id.toString(),
+        environmentId: workflowPreference._environmentId.toString(),
+        organizationId: workflowPreference._organizationId.toString(),
+        preferences: buildWorkflowPreferencesFromPreferenceChannels(
+          workflowPreference.critical,
+          workflowPreference.preferenceSettings
+        ),
+      });
+
+      await upsertPreferences.upsertUserWorkflowPreferences(userWorkflowPreferenceToUpsert);
+
+      const workflowPreferenceToUpsert = UpsertWorkflowPreferencesCommand.create({
+        templateId: workflowPreference._id.toString(),
+        environmentId: workflowPreference._environmentId.toString(),
+        organizationId: workflowPreference._organizationId.toString(),
+        preferences: DEFAULT_WORKFLOW_PREFERENCES,
+      });
+
+      await upsertPreferences.upsertWorkflowPreferences(workflowPreferenceToUpsert);
```

Comment:
Am I right to assume that V1 workflows that it is the majority of this migration, only have user workflow preferences?